src/abstractdatatable/reshape.jl

```diff
@@ -193,30 +188,17 @@ function unstack(dt::AbstractDataTable, rowkey::Int, colkey::Int, value::Int)
     # `rowkey` integer indicating which column to place along rows
     # `colkey` integer indicating which column to place along column headers
     # `value` integer indicating which column has values
-    refkeycol = NullableCategoricalArray(dt[rowkey])
```

Comment:
No, the problem is deeper than that:
```julia
julia> dt = DataTable(id=1:2, variable=["a", "b"], value=3:4)
2×3 DataTables.DataTable
│ Row │ id │ variable │ value │
├─────┼────┼──────────┼───────┤
│ 1   │ 1  │ a        │ 3     │
│ 2   │ 2  │ b        │ 4     │

julia> unstack(dt, :id, :variable, :value)
2×3 DataTables.DataTable
│ Row │ id │ a     │ b     │
├─────┼────┼───────┼───────┤
│ 1   │ 1  │ 3     │ #NULL │
│ 2   │ 2  │ #NULL │ 4     │

```

You can never know in advance whether there will be a row for each cell to fill.